extensions/WindowsSandboxExtension/Helpers/DependencyChecker.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Management;
+using System.Security.Principal;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Management.Deployment;
+using WinRT;
+
+namespace WindowsSandboxExtension.Helpers;
+
+internal sealed class DependencyChecker
+{
+    private const string OptionalComponentName = "Containers-DisposableClientVM";
+    private const string PackageFamilyName = "MicrosoftWindows.WindowsSandbox_cw5n1h2txyewy";
+
+    public static bool IsOptionalComponentEnabled()
```

Comment:
Method should check if the optional component is Absent as well, since Sandbox isn't available on Home skus.

Reply:
This returns if the optional component is absent because the Managed Object would not find any features and the foreach loop would terminate. For the purposes, absent is a subset of not enabled.

Reply:
fair I guess I was thinking in terms of logging that at least you'll have it in the logs that the machine doesn't support.

Reply:
I see. I can some logging here.

Reply:
Although maybe it's not a great idea to check the SKU because it is possible that in the future, we bring Windows Sandbox to other SKUs. I think it's better to just logged that its not enabled (which I already do). And the users can go check MSDN to see how to enable. That will have a comprehensive list of SKUs and Windows version numbers in which it is supported.